File: 프로그래머스/unrated/152995. 인사고과/인사고과.js
function solution(scores) {
  let answer = 1;
  const target = scores[0];

  scores.sort((a, b) => {
    if (a[0] !== b[0]) return b[0] - a[0];
    return a[1] - b[1];
  });

  let before = 0;
  for (const s of scores) {
    if (target[0] < s[0] && target[1] < s[1]) return -1;

    if (before <= s[1]) {
      if (target[0] + target[1] < s[0] + s[1]) answer++;
      before = s[1];
    }
  }
  return answer;
}

File: 프로그래머스/unrated/152995. 인사고과/인사고과.py
def solution(scores):
    wonho = scores[0]
    
    scores.sort(key=lambda x: (-x[0], x[1]))
    
    curr = scores[0]
    possible_list = []
    for score in scores:
        curr = [max(curr[0], score[0]), max(curr[1], score[1])]
        if score[0] < curr[0] and score[1] < curr[1]:
            if score == wonho:
                return -1
            continue
        possible_list.append(score)
            
    temp = sorted(possible_list, key=lambda x : sum(x))
    sorted_list = temp[::-1]

    if wonho == sorted_list[0]:
        return 1
    
    rank = [1]
    for i in range(1, len(sorted_list)):
        if sum(sorted_list[i]) == sum(sorted_list[i-1]):
            rank.append(rank[-1])
        else:
            rank.append(i+1)
            
        if sorted_list[i] == wonho:
            return rank[-1]
        
    
        